### PR TITLE
fix map assignment to entry in nil map,when use --feature-gates in kubeadm

### LINKED ISSUE #82978: kubeadm init --feature-gates panic
**What happened**:
kubeadm init --feature-gates get panic 
image

**What you expected to happen**:
no panic
**How to reproduce it (as minimally and precisely as possible)**:
kubeadm init --feature-gates
**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by 41tair**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:

/kind bug

**What this PR does / why we need it**:
fix kubeadm args --feature-gates bug
**Which issue(s) this PR fixes**:

Fixes #82978

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/componentconfigs/defaults.go [hunk 1]
```diff
@@ -59,6 +59,7 @@ const (
 // DefaultKubeProxyConfiguration assigns default values for the kube-proxy ComponentConfig
 func DefaultKubeProxyConfiguration(internalcfg *kubeadmapi.ClusterConfiguration) {
 	externalproxycfg := &kubeproxyconfigv1alpha1.KubeProxyConfiguration{}
+	externalproxycfg.FeatureGates = make(map[string]bool)
```
Comment (dixudx) [inline cmd/kubeadm/app/componentconfigs/defaults.go]: could use

```go
externalproxycfg := &kubeproxyconfigv1alpha1.KubeProxyConfiguration{FeatureGates: make(map[string]bool)}
```

**Comment by dixudx** [comment]
@41tair Please squash your commits as well. Thanks for your contribution.

**Comment by 41tair** [comment]
@dixudx fix and squash commit

**Comment by neolit123** [comment]
@41tair please provide more details on what sort of bug you are seeing and steps to reproduce.

**Comment by 41tair** [comment]
@neolit123 I added a related issue #82978

**Comment by neolit123** [comment]
@41tair i didn't see a version in your issue ticket, but i'm going to assume 1.16. thanks for providing the panic output.

/lgtm
/approve
/ok-to-test
/priority critical-urgent

/ok-to-test
cc @Arvinderpal 
we need to backport this for 1.16!

**Comment by neolit123** [comment]
@kubernetes/sig-cluster-lifecycle-pr-reviews

**Pull request merged**